add initial scribe CLI with examples

sallyom opened the pull request:
Signed-off-by: Sally O'Malley <somalley@redhat.com>

**Describe what this PR does**
Adds `scribe` command line interface 

**Is there anything that requires special attention?**
Will be actively developing after initial merge

**TODO in followup**
* add unit-tests
* add a list of TODOs
* add full sync command

JohnStrunk commented on pkg/cmd/destination.go:
```diff
@@ -0,0 +1,252 @@
+)
+
+var (
+	scribeNewDestinationLong = templates.LongDesc(`
```
These come out as pretty big walls of text. Could they be line wrapped like https://github.com/kubernetes/kubectl/blob/0149779a03735a5d483115ca4220a7b6c861430c/pkg/cmd/get/get.go#L88

sallyom replied:
will do

sallyom replied:
done

JohnStrunk commented on cmd/scribe/scribe.go:
```diff
@@ -0,0 +1,45 @@
+	if flag := from.Lookup("v"); flag != nil {
+		level := flag.Value.(*klog.Level)
+		levelPtr := (*int32)(level)
+		flags.Int32Var(levelPtr, "loglevel", 0, "Set the level of log output (0-10)")
```
Do we need both of these?
```
      --loglevel int32                 Set the level of log output (0-10)
      --v int32                        Set the level of log output (0-10)
```

sallyom replied:
probably not, will update

sallyom replied:
fixed, have `-v --version` (bc that's defaulted w/ cobra afaict) then `--loglevel` for the verbosity - fixed the double verbosity set and also the ambiguous `-v` for version & loglevel

sallyom commented on .golangci.yml:
```diff
@@ -19,7 +19,7 @@ linters:
     - govet
     - ineffassign
-    - lll
+    #- lll scribe cli cobra command flag descriptions break this
```
oh great, ok will update

sallyom replied:
updated

sallyom commented on README.md:
```diff
@@ -15,24 +15,59 @@ Card](https://goreportcard.com/badge/github.com/backube/scribe)](https://gorepor
+Follow the steps in the [installation
 instructions](https://scribe-replication.readthedocs.io/en/latest/installation/index.html).
+Here are
+[useful commands to configure cluster storage classes](https://github.com/backube/scribe/blob/master/storage-setup.md).
```
done

sallyom commented on README.md:
```diff
@@ -15,24 +15,59 @@ Card](https://goreportcard.com/badge/github.com/backube/scribe)](https://gorepor
 ## Getting started
 
-To try out Scribe,  follow the steps in the [installation
+### Try Scribe in Kind
```
updated, ptal

sallyom commented on pkg/cmd/options.go:
```diff
@@ -0,0 +1,149 @@
+
+func (o *ReplicationOptions) getCopyMethod(c *CommonOptions) (*CommonOptions, bool) {
+	// CopyMethod is always required
+	switch o.CopyMethod {
```
updated

sallyom commented on pkg/cmd/options.go:
```diff
@@ -0,0 +1,149 @@
+	// Capacity not always required
+	switch {
+	case len(o.Capacity) > 0:
+		capacity := resource.MustParse(o.Capacity)
```
panic. there's probably a return err alternative, this is how i've always seen it run, though. i'll take a look

sallyom commented on pkg/cmd/options.go:
```diff
@@ -0,0 +1,149 @@
+	if len(o.ProviderParameters) > 0 {
+		p := strings.Split(o.ProviderParameters, ",")
+		for _, kv := range p {
+			pair := strings.Split(kv, "/")
```
done, updated

sallyom commented on storage-setup.md:
```diff
@@ -0,0 +1,43 @@
+# Tips for setting up storage in a K8s, OpenShift cluster
```
added! (to docs/installation/index.rst)

sallyom commented:
@JohnStrunk I addressed your fb. Also, I updated the examples to be within the docs, and *.rst.  I've verified the 2 examples I added are working, too. Thanks for your review!
